src/wallet/scriptpubkeyman.cpp

```diff
@@ -1635,4 +1635,212 @@ bool DescriptorScriptPubKeyMan::CanUpdateToWalletDescriptor(const WalletDescript
 
     return true;
 }
+
+SilentPaymentDescriptorScriptPubKeyMan::SilentPaymentDescriptorScriptPubKeyMan(WalletStorage& storage, WalletDescriptor& descriptor)
+    : DescriptorScriptPubKeyMan(storage, descriptor, 0)
+{
+    PopulateLabelTweaks();
+}
+
+void SilentPaymentDescriptorScriptPubKeyMan::PopulateLabelTweaks()
+{
+    LOCK(cs_desc_man);
+    if (m_wallet_descriptor.descriptor->GetOutputType() != OutputType::SILENT_PAYMENTS) {
+        throw std::runtime_error(std::string(__func__) + ": descriptor is not a Silent Payment Descriptor");
+    }
+    const auto provider{GetSPProvider()};
+
+    // Populate m_map_label_tweaks
+    auto change_label_data = bip352::CreateLabelTweak(provider.sp_keys.first, 0);
+    m_map_label_tweaks.insert(change_label_data);
+    for (int i = 1; i < m_wallet_descriptor.next_index; i++) {
+        // Add the other generated labelled destinations
+        m_map_label_tweaks.insert(bip352::CreateLabelTweak(provider.sp_keys.first, i));
+    }
```

Comment:
I'd remove this (and any other code) related to non-change labels from this PR. Ideally, implement just enough for labels to get change working, and then we can fully implement labels in a follow-up PR.